File: app/src/main/java/com/example/waveplayer/fragments/fragment_loading/ViewModelFragmentLoading.java
package com.example.waveplayer.fragments.fragment_loading;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.waveplayer.media_controller.MediaController;
import com.example.waveplayer.media_controller.MediaData;

public class ViewModelFragmentLoading extends ViewModel {

    private final MediaData mediaData = MediaData.getInstance();

    public LiveData<Double> getLoadingProgress() {
        return mediaData.getLoadingProgress();
    }

    public LiveData<String> getLoadingText() {
        return mediaData.getLoadingText();
    }

}

File: app/src/main/java/com/example/pinky_player/media_controller/ViewModelFragmentLoading.kt
package com.example.pinky_player.media_controller

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class ViewModelFragmentLoading(application: Application) : AndroidViewModel(application) {

    private val mediaData: MediaData = MediaData.getInstance(application.applicationContext)

    fun getLoadingProgress(): LiveData<Double> {
        return mediaData.loadingProgress
    }

    fun getLoadingText(): LiveData<String> {
        return mediaData.loadingText
    }

}